lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,36 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def respond_to?(name, include_private = false)
+        Injector.instance_methods.include?(name) || builder.respond_to?(name)
```

Comment:
Ahh OK, not used to inheriting BasicObject myself so don't know the interface really.

I'd probably remove it in that case, if it quacks like a duck...